edit-post/store/selectors.js

```diff
@@ -95,20 +94,6 @@ export function isPublishSidebarOpened( state ) {
 	return state.publishSidebarActive;
 }
 
-/**
- * Returns true if there's any open sidebar (mobile, desktop or publish).
- *
- * @param {Object} state Global application state.
- *
- * @return {boolean} Whether sidebar is open.
- */
-export function hasOpenSidebar( state ) {
-	const generalSidebarOpen = getPreference( state, 'activeGeneralSidebar' ) !== null;
-	const publishSidebarOpen = state.publishSidebarActive;
-
-	return generalSidebarOpen || publishSidebarOpen;
```

Comment:
We are still using a specific flag for publish sidebar, should we consolidate the behaviors and remove the flag from state?

Another point should we have a flag that tells if sidebars are open, and one property that tells the current sidebar that is open or will be open if sidebar flag is toggled?
The advantages being we can have a button to close and open the last sidebar, without resetting the last sidebar to close it. We would also be able to have open flags for different view sizes allowing us to recover the mobile behaviour where mobile sidebars open and close state is totally independent from desktop one.

Reply:
> We are still using a specific flag for publish sidebar, should we consolidate the behaviors and remove the flag from state?

That would be awesome. However, they have different UI. I didn't want to touch it as part of this PR, but rather focus on the code that might remove some code duplication between the editor and plugins.

> Another point should we have a flag that tells if sidebars are open, and one property that tells the current sidebar that is open or will be open if sidebar flag is toggled?

This might be tricky because of how you interact with the editor. When you open the right side menu of the block you want to see the advanced setting of the block. When you click on the cog button in the header you rather expect to see the document settings, but definitely not the last open plugin. So I would say toggling for the last open sidebar might be not the best fit in many cases. Did you have something different in mind? I might not understand your reasoning correctly. This thing is really tricky because of so many factors involved :)

Reply:
> When you click on the cog button in the header you rather expect to see the document settings, but definitely not the last open plugin.

Yes, that's true, but in that case, the cog button whould trigger, an action to change the sidebar to the general one and another action to open the sidebar.

I think my basic idea is the following we have one more flag (e.g: mobile and desktop) that say if a sidebar is opened or not. So we don't have a checking logic where we check the general sidebar, the plugins and publish.
And one field with the sidebar id of the sidebar that if sidebars are shown should be rendered.
The sidebar components would just check the flag to see if sidebars are opened and the id of the sidebar to check if they should render or not.

But, that is something we can do after e.g: as part of a mobile improvement, or as part of a refactor of our own sidebars to use the extensibility mechanism.

Reply:
Yes, we should definitely explore that further. Thanks for sharing your ideas 👍